### PR TITLE
DOC: Mention uniform in the np.random.Generator.random function.

### PR DESCRIPTION
**Pull request by bramton**
Added a _See also_ section to mention np.random.Generator.uniform.
``[ci skip]``

### CODE REVIEW

**Comment by charris** [comment]
`uniform` is a RandomState function. It would be better to recommend the `uniform` method of the rng being used. @rkern Thoughts?

**Comment by rkern** [comment]
I think that was their intention. It's just a matter of using the right reST/Sphinx syntax(es) to make it reference `np.random.Generator.uniform` in those two places. I always have to look it up.

File: numpy/random/_generator.pyx [hunk 1]
```diff
@@ -269,7 +269,7 @@ cdef class Generator:
 
         See Also
         --------
-        ~.Generator.uniform : Drawn samples from the parameterized uniform distribution.
+        `~.Generator.uniform` : Drawn samples from the parameterized uniform distribution.
```
Comment (rossbar) [inline numpy/random/_generator.pyx:272]: numpydoc does some pre-parsing on the See Also entries and can't currently handle things that start with `~` I believe.

I think it'll work correctly if you provide an explicit role, e.g.

```suggestion
        :meth:`~numpy.random.Generator.uniform` : Drawn samples from the parameterized uniform distribution.
```

That'll look good and link correctly in the html docs, but the source rst is kind of ugly so YMMV

**Comment by bashtage** [comment]
You don't need any special markup to get HTML links within a class

This 

https://github.com/numpy/numpy/blob/f3935c9dc9a661bf818a09e46da6497220bd7dca/numpy/random/_generator.pyx#L996-L1000

produces

See [https://numpy.org/doc/stable/reference/random/generated/numpy.random.Generator.standard_normal.html#numpy.random.Generator.standard_normal](standard_normal)

**Comment by charris** [comment]
Thanks @bramton .

**Pull request merged**